### PR TITLE
feat: onboard webengage web device mode

### PR DESCRIPTION
**Pull request by yashasvibajpai**
## PR Description

Resolves INT-291
Onboarding Webengage web device mode destination.

## Linear task (optional)

Linear task link

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Comment by bardisg** [comment]
Can we please add unit tests for these new additions?

File: packages/analytics-js-integrations/src/integrations/Webengage/browser.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable class-methods-use-this */
+import { logger } from '@rudderstack/analytics-js-common/v1.1/utils/logUtil';
```
Comment (bardisg) [inline packages/analytics-js-integrations/src/integrations/Webengage/browser.js:2]: use the new logger as per the https://github.com/rudderlabs/rudder-sdk-js/pull/1434 PR

Comment (bardisg) [reply packages/analytics-js-integrations/src/integrations/Webengage/browser.js:2]: same goes for all logger calls, follow same conventions on how logger is utilised

File: packages/analytics-js-integrations/src/integrations/Webengage/nativeSdkLoader.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { LOAD_ORIGIN } from '@rudderstack/analytics-js-common/v1.1/utils/constants';
+
+function loadNativeSdk(licenseCode, dataCentre, window) {
+  var webengage;
+  !(function (w, e, b, n, g) {
```
Comment (bardisg) [inline packages/analytics-js-integrations/src/integrations/Webengage/nativeSdkLoader.js:5]: add eslint and sonar start/stop magic comments to exclude the loading snippet of the third party from checks

**Review by bardisg — CHANGES_REQUESTED** [review]

File: packages/analytics-js-integrations/src/integrations/Webengage/nativeSdkLoader.js (hunk 2 above)
Comment (saikumarrs) [reply packages/analytics-js-integrations/src/integrations/Webengage/nativeSdkLoader.js:5]: @bardisg These statements are not needed here as all `nativeSdkLoader.js` files are excluded in the config.

**Pull request closed without merging**